tests/test_docker_cat.py

```diff
@@ -0,0 +1,539 @@
+
+    def test_check_qg(self):
+        """As a SonarQube user, I want the SonarQube server to have the CNES
+        Quality Gate configured and set as default so that I can use it.
+        """
+        quality_gates = requests.get(f"{self.CAT_URL}/api/qualitygates/list").json()['qualitygates']
+        cnes_quality_gates = [ gate for gate in quality_gates if gate['name'] == "CNES" ]
+        # Hint: if one of these tests fails, the CNES Quality Gate may not have been added correctly, check the container logs
+        assert cnes_quality_gates # not empty
+        assert cnes_quality_gates[0]['isDefault']
+
+    def test_check_qp(self):
+        """As a SonarQube user, I want the SonarQube server to have the
+        CNES Quality Profiles available so that I can use them.
+        """
+        required_quality_profiles = (
+            "CNES_JAVA_A",
+            "CNES_JAVA_B",
+            "CNES_JAVA_C",
+            "CNES_JAVA_D",
+            "CNES_PYTHON_A",
+            "CNES_PYTHON_B",
+            "CNES_PYTHON_C",
+            "CNES_PYTHON_D",
+            "CNES_CPP_A",
+            "CNES_CPP_B",
+            "CNES_CPP_C",
+            "CNES_CPP_D",
+            "CNES_C_A",
+            "CNES_C_B",
+            "CNES_C_C",
+            "CNES_C_D",
+            "CNES_C_EMBEDDED_A",
+            "CNES_C_EMBEDDED_B",
+            "CNES_C_EMBEDDED_C",
+            "CNES_C_EMBEDDED_D"
+        )
+        quality_profiles = requests.get(f"{self.CAT_URL}/api/qualityprofiles/search").json()['profiles']
+        cnes_quality_profiles = [ qp['name'] for qp in quality_profiles if re.match(r'CNES_\w+_[ABCD]', qp['name']) ]
+        for profile in required_quality_profiles:
+            # Hint: if this test fails, one or more Quality Profiles may be missing, check the container logs
+            assert profile in cnes_quality_profiles
+
+    # Language tests
+    def test_language_c_cpp(self):
+        """As a user of this image, I want to analyze a C/C++ project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        sensors = (
+            "Sensor C++ (Community) SquidSensor [cxx]",
+            "Sensor SonarFrama-C [framac]"
+        )
+        self.language("C/C++", "c++", "c_cpp", sensors, "c-dummy-project", 0, "CNES_C_A", 1)
+        # 0 issue are expected with the Sonar way Quality Profile for
+        # C++ (Community) because it does not have any rule enabled.
+    
+    def test_language_fortran_77(self):
+        """As a user of this image, I want to analyze a fortran 77 project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        sensors = (
+            "INFO: Sensor Sonar i-Code [icode]",
+        )
+        self.language("Fortran 77", "f77", "fortran77", sensors, "fortran77-dummy-project", 11)
+
+    def test_language_fortran_90(self):
+        """As a user of this image, I want to analyze a fortran 90 project
```

Comment:
Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/docker-cat/pullRequest?prid=6253839)